src/openrct2/paint/tile_element/Paint.Surface.cpp

```diff
@@ -360,24 +339,22 @@ static ImageId GetEdgeImage(const TerrainEdgeObject* edgeObject, uint8_t type)
     return result;
 }
 
-static ImageId GetTunnelImage(const TerrainEdgeObject* edgeObject, uint8_t type, edge_t edge)
+static ImageId GetTunnelImage(const TerrainEdgeObject* edgeObject, TunnelType type, edge_t edge)
 {
-    static constexpr uint32_t offsets[TUNNEL_TYPE_COUNT] = { 36, 40, 44, 48, 52, 56, 60, 64, 68, 72, 76, 80,
-                                                             36, 48, 60, 72, 76, 80, 84, 88, 92, 96, 100 };
-
     bool hasDoors = false;
     if (edgeObject != nullptr)
     {
         hasDoors = edgeObject->HasDoors && !edgeObject->UsesFallbackImages();
     }
 
-    if (!hasDoors && type >= REGULAR_TUNNEL_TYPE_COUNT && type < std::size(offsets))
-        type = TUNNEL_0;
+    if (!hasDoors && EnumValue(type) >= kRegularTunnelTypeCount && EnumValue(type) < std::size(kTunnels))
+        type = TunnelType::StandardFlat;
```

Comment:
Given we now have TunnelType being a strong enum I think we should remove this check and rather make sure we never pass down something wrong, I looked at the usage and I don't think its possible to have something odd here.

Reply:
Checking myself and came to the same conclusion. Removed the bounds check.